code/modules/mob/living/basic/lavaland/goldgrub/goldgrub.dm

```diff
@@ -0,0 +1,173 @@
+	icon = 'icons/mob/simple/lavaland/lavaland_monsters_wide.dmi'
+	icon_state = "goldgrub"
+	icon_living = "goldgrub"
+	icon_dead = "goldgrub_dead"
+	icon_gib = "syndicate_gib"
+	speed = 5
+	pixel_x = -12
+	base_pixel_x = -12
+	mob_biotypes = MOB_ORGANIC|MOB_BEAST
+	friendly_verb_continuous = "harmlessly rolls into"
+	friendly_verb_simple = "harmlessly roll into"
+	maxHealth = 45
+	health = 45
+	melee_damage_lower = 0
+	melee_damage_upper = 0
+	attack_verb_continuous = "barrels into"
+	attack_verb_simple = "barrel into"
+	attack_sound = 'sound/weapons/punch1.ogg'
+	combat_mode = FALSE
+	speak_emote = list("screeches")
+	death_message = "stops moving as green liquid oozes from the carcass!"
+	status_flags = CANPUSH
+	gold_core_spawnable = HOSTILE_SPAWN
+	ai_controller = /datum/ai_controller/basic_controller/goldgrub
+	///can this mob lay eggs
+	var/can_lay_eggs = TRUE
+	///can we tame this mob
+	var/can_tame = TRUE
+	//pet commands when we tame the grub
+	var/list/pet_commands = list(
+		/datum/pet_command/idle,
+		/datum/pet_command/free,
+		/datum/pet_command/grub_spit,
+		/datum/pet_command/follow,
+		/datum/pet_command/point_targetting/fetch
+	)
+
+/mob/living/basic/mining/goldgrub/Initialize(mapload)
+	. = ..()
+	if(can_tame)
+		make_tameable()
+	if(can_lay_eggs)
+		make_egg_layer()
+	generate_loot()
+	var/datum/action/cooldown/mob_cooldown/spit_ore/spit = new(src)
+	var/datum/action/cooldown/mob_cooldown/burrow/burrow = new(src)
+	AddComponent(/datum/component/appearance_on_aggro, overlay_icon = icon, overlay_state = "[icon_state]_alert")
+	AddElement(/datum/element/wall_smasher)
+	AddComponent(/datum/component/ai_listen_to_weather)
+	spit.Grant(src)
+	burrow.Grant(src)
+	ai_controller.set_blackboard_key(BB_SPIT_ABILITY, spit)
+	ai_controller.set_blackboard_key(BB_BURROW_ABILITY, burrow)
+
+/mob/living/basic/mining/goldgrub/UnarmedAttack(atom/attack_target, proximity_flag, list/modifiers)
+	. = ..()
+	if(!.)
+		return
+
+	if(!proximity_flag)
+		return
+
+	if(istype(attack_target, /obj/item/stack/ore))
+		consume_ore(attack_target)
+
+/mob/living/basic/mining/goldgrub/bullet_act(obj/projectile/bullet)
+	if(stat == DEAD)
+		return BULLET_ACT_FORCE_PIERCE
+	else
+		visible_message(span_danger("The [bullet.name] is repelled by [src]'s girth!"))
+		return BULLET_ACT_BLOCK
+
+/mob/living/basic/mining/goldgrub/proc/barf_contents()
+	visible_message(span_danger("[src] spits out its consumed ores!"))
+	playsound(src, 'sound/effects/splat.ogg', 50, TRUE)
+	for(var/obj/item/ore as anything in src)
+		ore.forceMove(loc)
+
+/mob/living/basic/mining/goldgrub/proc/generate_loot()
+	var/loot_amount = rand(1,3)
+	var/list/weight_lootdrops = list(
+		/obj/item/stack/ore/silver = 4,
+		/obj/item/stack/ore/gold = 3,
+		/obj/item/stack/ore/uranium = 3,
+		/obj/item/stack/ore/diamond = 1,
+	)
+	var/list/death_loot = list()
+	for(var/i in 1 to loot_amount)
+		death_loot += pick_weight(weight_lootdrops)
+	AddElement(/datum/element/death_drops, death_loot)
```

Comment:
Instead of using death drops, which is not really intended to have random loot pools within, just tie it into the fact that we already consume ore and drop it on death via barf_contents

```suggestion
	for(var/i in 1 to loot_amount)
		var/picked_goodie = pick_weight(weight_lootdrops)
		new picked_goodie(src) // starts us off with "consumed ore" that we loose via barf_contents if we die
```